fuel-vm/fuzz/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+use fuel_vm::fuel_asm::op;
+use fuel_vm::fuel_asm::{Instruction, InvalidOpcode};
+use fuel_vm::fuel_types::Word;
+use fuel_vm::prelude::field::Script;
+use fuel_vm::prelude::*;
+
+use fuel_vm::util::test_helpers::TestBuilder;
+use fuel_vm::{fuel_asm, script_with_data_offset};
+use fuel_vm::fuel_types::canonical::Serialize;
+use std::ops::Range;
+
+/// Magic value used as separator between fuzz data components in corpus files.
+const MAGIC_VALUE_SEPARATOR: [u8; 8] = [0x00u8, 0xAD, 0xBE, 0xEF, 0x55, 0x66, 0xCE, 0xAA];
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct FuzzData {
+    pub program: Vec<u8>,
+    pub sub_program: Vec<u8>,
+    pub script_data: Vec<u8>,
+}
+
+pub fn encode(data: &FuzzData) -> Vec<u8> {
+    let seperator: Vec<_> = MAGIC_VALUE_SEPARATOR.into();
+    let parts: [Vec<u8>; 5] = [
+        data.program.iter().copied().collect(),
+        seperator.clone(),
+        data.script_data.clone(),
+        seperator.clone(),
+        data.sub_program.iter().copied().collect(),
+    ];
+
+    let data: Vec<u8> = parts.iter().flatten().copied().collect();
+    data
+}
+
+fn split_by_separator(data: &[u8], separator: &[u8]) -> Vec<Range<usize>> {
```

Comment:
Is it possible to replace this function by https://doc.rust-lang.org/std/primitive.slice.html#method.split ?